runner-plugins/runner-scribe/main.go

```diff
@@ -209,6 +256,94 @@ func makeVulnerability(initems []gozdef.VulnEvent, cmd mig.Command) (items []goz
 	return
 }
 
+// given config for an API behind Auth0 (including client ID and Secret), 
+// return an Auth0 access token beginning with "Bearer "
+// pattern from https://auth0.com/docs/api-auth/tutorials/client-credentials
+func GetAuthToken(api ServiceApi) (authToken string) {
+	payload := strings.NewReader("{\"grant_type\":\"client_credentials\",\"client_id\": \"" + api.ClientID + "\",\"client_secret\": \"" + api.ClientSecret + "\",\"audience\": \"" + api.URL + "\"}")
+	req, _ := http.NewRequest("POST", api.AuthEndpoint, payload)
+	req.Header.Add("content-type", "application/json")
+
+	res, err := http.DefaultClient.Do(req)
+	if err != nil {
+		panic(err)
+	}
+
+	defer res.Body.Close()
+	bodyJSON, _ := ioutil.ReadAll(res.Body)
+	
+	// unpack the JSON into an Auth0 token struct
+	var body Auth0Token
+	err = json.Unmarshal(bodyJSON, &body)
+	if err != nil {
+		panic(err)
+	}
+
+	// serviceAPI expects the Access token in the form of "Bearer <token>"
+	authToken = "Bearer " + body.Access_token
+	return
+}
+
+// query a ServiceAPI instance for the set of all assets
+// load them into a searchable map, keyed to asset hostname
+// the ServiceAPI object must already be loaded with a Bearer token
+func GetAssets(m map[string]ServiceApiAsset, api ServiceApi) (err error){
+	
+	// get json array of assets from serviceapi
+	requestURL := api.URL + "api/v1/assets/"
+	req, err := http.NewRequest(http.MethodGet, requestURL, nil)
+	if err != nil {
+		return err
+	}
+	req.Header.Add("accept", "application/json")
+	req.Header.Add("Authorization", api.Token)
+	res, err := http.DefaultClient.Do(req)
+	if err != nil {
+		return err
+	}
+
+	// unpack the HTTP request response
+	defer res.Body.Close()
+	body, readErr := ioutil.ReadAll(res.Body)
```

Comment:
Rather than reading the entire response body into memory before trying to unmarshal it from JSON, we should use an [encoding/json.Decoder](https://golang.org/pkg/encoding/json/#Decoder).

Reply:
I worked on this for a bit, but there is something weird about how ServiceAPI is presenting back the JSON, requiring double-unmarshaling. I couldn't get it to come in with decoding. 

I'd like to defer this change, since we control both ends of the information. Maybe a fix on the serviceAPI side would solve it easier.